packages/focusvisible/src/FocusVisibleContainer.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import PropTypes from 'prop-types';
+
+import { useFocusVisible } from './useFocusVisible';
+
+export function FocusVisibleContainer({
+  scope,
+  relativeDocument,
+  className,
+  dataAttribute,
+  children
+}) {
+  useFocusVisible({ scope, relativeDocument, className, dataAttribute });
```

Comment:
If you look at the other hooks I don't bother with destructuring the props here and just pass through as `...options`. Instead I just define the props in propTypes as you have done.

I know this is a different hook compared to the other in that it doesn't actually return anything but it should still be used the same was as the others e.g.

```js
function FocusVisibleContainer({children, render = children, ...options}) {
  return render(useFocusVisible(options));
}
```

Then the container is used like all the other render prop containers:

```js
<FocusVisibleContainer scope={ref}>
  {() => (
    <div ref={ref} data-test-id="wrapper">
      <button data-test-id="button" tabIndex="0"></button>
      <input data-test-id="input" />
      <textarea data-test-id="textarea"></textarea>
    </div>
  )
</FocusVisibleContainer>
```

Reply:
> If you look at the other hooks I don't bother with destructuring the props here and just pass through as ...options. Instead I just define the props in propTypes as you have done.

Fixed.

> Then the container is used like all the other render prop containers:

This made me realize the `scope/ref` placement didn't make sense in the context of render-props. I've now updated the `FocusVisibleContainer` to create the ref internally and provide it to the render prop. The updated usage is now:

```jsx
<FocusVisibleContainer>
        {({ ref }) => (
          <div ref={ref} data-test-id="wrapper">
            <button data-test-id="button"></button>
            <input data-test-id="input" />
            <textarea data-test-id="textarea"></textarea>
          </div>
        )}
      </FocusVisibleContainer>
```